core/node/rpc/add_event.go

```diff
@@ -93,11 +103,19 @@ func (s *Service) addParsedEvent(
 					return err
 				}
 			}
+			var chainAuthArgsListStringBuilder strings.Builder
+			chainAuthArgsListStringBuilder.WriteString("[\n")
+			for _, chainAuthArgs := range chainAuthArgsList {
+				chainAuthArgsListStringBuilder.WriteString("  ")
+				chainAuthArgsListStringBuilder.WriteString(chainAuthArgs.String())
+				chainAuthArgsListStringBuilder.WriteString(",\n")
+			}
+			chainAuthArgsListStringBuilder.WriteString("]")
```

Comment:
Maybe there is a good reason for but this looks a bit fishy. Why not leave the formatting to the logger?

Reply:
You know, I never did figure out how toString works with go - I think you're right, there's already a string method on ChainAuthArgs so this is probably unnecessary.